### ISSUE TITLE
When saving and retrieving the data in the editor iam seeing extra space

### ISSUE THREAD

**Issue by vibingopal** [KillerCodeMonkey/ngx-quill#151]
Hi,

When i try to save the contents of the editor using Quilleditor and retrieving as per document, iam seeing extra space every time.

Below is my screenshot of my code and output:

Code:
![image](https://user-images.githubusercontent.com/17860925/38177391-4197a032-35c6-11e8-96af-19bae4edb181.png)

Output:
![image](https://user-images.githubusercontent.com/17860925/38177396-6d03e3ac-35c6-11e8-8ebb-873f72a87f73.png)

![image](https://user-images.githubusercontent.com/17860925/38177402-87e7d318-35c6-11e8-9e44-32a4652f9683.png)

![image](https://user-images.githubusercontent.com/17860925/38177430-e2451bfe-35c6-11e8-99a4-628922e927ad.png)

**Comment by vibingopal** [comment]
Any update on this please ?

**Comment by KillerCodeMonkey** [comment]
No time to check this ;). But i think i saw a similar issue in the quilljs
repo.

Please check this first. I die not change the default quill behavior.

On 2 Apr 2018 20:17, "vibingopal" <notifications@github.com> wrote:

Any update on this please ?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/KillerCodeMonkey/ngx-quill/issues/151#issuecomment-377999621>,
or mute the thread

**Comment by KillerCodeMonkey** [comment]
ehhm do you have a working plunkr or something like this?
I can not see any model or formControl connected to your editor. How do you set the initial value?

**Comment by vibingopal** [comment]
Thanks for the response. I am setting the initial value of my editor in the placeholder first (Something like Add your story...).

Later user can enter as per shown in the screen shot..As soon as iam saving the content after storing.. In the console log iam seeing extra spaces like above. Not able to find the existing same issue as suggested in the issues tracker. could you please suggest what went wrong?

As soon as user clicked on save after entering the data, iam calling a method like below and printing the console output..


>`  let quill = new Quill("#editor-container", { } )
     let delta = quill.getContents();
     console.log("Value is",delta)
`

**Comment by KillerCodeMonkey** [comment]
yeah but if the problem also exists in plain quilljs, this is totally a quilljs issue and not a ngx-quill one.

And i can not see in your screenshot how the value is saved or the saved value is passed to the quill-editor-component ;).

if you check out my demo repo and check the JS-console there is a listener for the change event and there are no extra spaces, so i think it is a problem how you are using the editor and my component.

**Comment by vibingopal** [comment]
As soon as user clicked on save button iam passing an event to parent component like below 

![image](https://user-images.githubusercontent.com/17860925/38308103-a00fad32-37db-11e8-8158-4fc8ee0673d4.png)

And retrieving the value of the event in the editor to display. Anyways in the console log itself iam seeing extra lines before the data get popped up in the quill-editor component

 ```
retrieveData(event) {
    let quill = new Quill("#editor", {});
    quill.setContents(event.data);
  }
```

**Comment by KillerCodeMonkey** [comment]
Yeah but then you are not using nix-quill anymore. And it is not a ngx-quill issue ;)

I am not a quilljs expert. sorry

**Comment by vibingopal** [comment]
i didn't understand that iam not using ngx-quill. I have used quill-editor as per the example provided in the ngx-quill repository. Not sure what iam not using

**Comment by KillerCodeMonkey** [comment]
Ehm you are creating your own quill instance in your screenshot? You do not do this if you are using ngx-quill, because it does it for you
